### ISSUE TITLE
Ignore 'Option Strict On' defined in option settings with .net 5 in VB.

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/project-system#7708]
_This issue has a corresponding [ticket on Developer Community](https://developercommunity.visualstudio.com/t/Ignore-Option-Strict-On-defined-in-opt/1535957). Please vote and comment there to make sure your voice is heard._

---
Hello,

With WPF and Winforms, using .net 5 and visual studio version 16.11.3, the Option Strict setting to On in the settings options is ignored.

It is necessary to add the instruction in the code.

Despite Option Strict On, there is no conversion error detected.

![image](https://user-images.githubusercontent.com/350947/138000133-c93ad5e9-0daa-4e95-93ae-4e68a765271d.png)

With the addition of the Option Strict On statement, the conversion error is detected.

![image](https://user-images.githubusercontent.com/350947/138000152-22866b30-186c-4601-971f-0cddf57e6c31.png)

**Comment by KlausLoeffelmann** [comment]
The dialog shown here is from the Option Dialog, which 'only' suggest the default settings for a new project. It does not change the current setting.

You change the current settings over the Project settings (Properties of Project in the Solution Exploreer):

![image](https://user-images.githubusercontent.com/9663150/149244376-e7b1f2f1-8721-4b7d-8b5c-0202bb61ad3e.png)


That setting I just tested, and it's working fine for both WPF and WinForms.

Closing this.

**Comment by Perpete** [comment]
Mr Klaus Loeffelman, your comment is correct.
But by performing a new test of creating a project, I just noticed that the default setting of Option Strict to 'On' in the options was not correctly taken into account when creating a new project.
You can see this discrepancy on the screenshot.
This is where the problem lies.

![Options](https://user-images.githubusercontent.com/66659162/180135613-1a69e97b-d874-4f0f-9f24-4b90451b0d49.png)

**Comment by Perpete** [comment]
Mr Klaus Löffelmann your comment is correct.

While performing a new test of creating a project, I have just noticed that the default setting of Option Strict to 'On' in the options was not taken correctly into accoun in the compilation properties after the creation of the new project.

You can see this discrepancy on the screenshot.
This is where the problem lies.

![Options](https://user-images.githubusercontent.com/66659162/180137797-8278d1be-a794-4916-842b-a11d2493e3ac.png)

**Comment by KlausLoeffelmann** [comment]
@melytc, do you think that is project system?
If yes, let's talk about this later this month. I have a couple of questions about this in addition.
Also @merriemcgaw, @KathleenDollard FYI.

**Comment by Perpete** [comment]
Hello,
Uncorrected issue with visual studio version 17.3.0.

**Comment by melytc** [comment]
Reopening issue to be investigated further. I can't remember where we left in this conversation but let's meet next week to discuss further questions @KlausLoeffelmann 🙂 

I tested this on VS 2022, using a VB WinForms project and I can still see the discrepancy on the strict option:

![image](https://github.com/dotnet/project-system/assets/8518253/3c309bb4-a323-4f43-90d8-e255bd1de768)

**Comment by KlausLoeffelmann** [comment]
Yes, let's do this please - this would be important to get fixed!
Thanks - just schedule any free spot you find but Monday morning! 😸
